### PR TITLE
Feat icon

### PR DESCRIPTION
**Pull request by Amber-Song**
# Context

Link to [Notion ticket](https://www.notion.so/cloudcannon/Icon-a3273cf7d4fe4d6d867332e1c93b36ef?pvs=4)

# Additional information

The icon doesn't work well for bookshop. To see how the icon is, add the front matter to pages like:
In `index.html`
```
    - _bookshop_name: generic/icon
      svg_icon: solid/academic-cap
      big_icon: false
      rounded_border: false
```
Then you can see the icon in index page in your local machine. You still not be able to see it in cloud cannon site because it uses bookshop, but you can see it in live site.

* Never mind: Above is changed.

# Testing (tester)

The reviewer should check on this branch:

- The code
- The component in index page
- The site in CloudCannon ([link here](https://vital-caterpillar.cloudvent.net/))

# Before merge (PR owner)

- Delete the test site in CloudCannon

### CODE REVIEW

File: src/pages/index.html [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
-content_blocks: 
-    - _bookshop_name: generic/sample 
-      text: Hello World!
-
+content_blocks:
+    -
+        _bookshop_name: generic/icon
+        svg_icon: outline/arrow-long-up
+        big_icon: true
+        rounded_border: true
```
Comment (aleksandrgordeev) [inline src/pages/index.html]: Maybe we should discard these changes? It looks like a test output.

File: src/pages/booking.html [hunk 2]
```diff
@@ -7,4 +7,4 @@
 content_blocks:
 ---
 
-<h1>booking</h1>
\ No newline at end of file
+<h1>booking</h1>
```
Comment (aleksandrgordeev) [inline src/pages/booking.html:10]: What is the purpose of this change?

File: component-library/components/generic/icon/icon.bookshop.yml [hunk 3]
```diff
@@ -0,0 +1,613 @@
...
+    - content_blocks
+  label: "Icon"
+  description: Icons
+  icon: bolt
+  tags: []
+
+# Defines the structure of this component, as well as the default values
+blueprint:
+  svg_icon:
+  big_icon: true
```
Comment (aleksandrgordeev) [inline component-library/components/generic/icon/icon.bookshop.yml]: This parameter is not in the ticket description. May we add this to the ticket description?

**Review by aleksandrgordeev — CHANGES_REQUESTED** [review]
The component works well 👍

I've added a couple of small suggestions. And there is a merge conflict that needs to be resolved.

File: src/pages/booking.html (hunk 2 above)
Comment (James-cloudcannon) [reply src/pages/booking.html:10]: this was just code from the boiler plate, looks like a commit removed it and then added back? These lines on each page can be deleted completely.

File: src/pages/index.html (hunk 1 above)
Comment (James-cloudcannon) [reply src/pages/index.html]: Yeah I assume this was just in place for testing the component. It makes it easier for us to see but we can just add this manually ourselves if instead of pushing this if needed.

File: component-library/components/generic/icon/icon.scss [hunk 4]
```diff
@@ -0,0 +1,24 @@
+.c-icon {
+    &__round {
+        border-radius: 50%;
+        border-color: var(--neutral-000, #FFF);
```
Comment (James-cloudcannon) [inline component-library/components/generic/icon/icon.scss]: --neutral-000 needs to be changed to --brand-grey-000

File: component-library/components/generic/icon/icon.scss
Comment (James-cloudcannon) [inline component-library/components/generic/icon/icon.scss:1]: Can't seem to get the border-color to work. As a simple workaround you can remove `border-color: transparent `out of __big and __small and adding something like this to each of those classes:

`&:not(.c-icon__round) {
            border-color: transparent;
        }`

File: component-library/components/generic/icon/icon.eleventy.liquid
Comment (James-cloudcannon) [inline component-library/components/generic/icon/icon.eleventy.liquid:1]: Seems to work well, assuming that the colour is inherited from the component using this.

File: component-library/components/generic/icon/icon.scss
Comment (James-cloudcannon) [inline component-library/components/generic/icon/icon.scss:1]: Might need to work on the padding & width/height. Small should be 32x32 pixels. Large should be 69x69 pixels

**Review by James-cloudcannon — COMMENTED** [review]
Nice work! just a few small changes to do and then we should be able to get this merged.

File: src/pages/index.html (hunk 1 above)
Comment (Amber-Song) [reply src/pages/index.html]: I will make sure to delete it before merge.

File: component-library/components/generic/icon/icon.scss
Comment (Amber-Song) [reply component-library/components/generic/icon/icon.scss:1]: Oh, that works! Amazing! I almost try to hack it with !important. 😅

File: component-library/components/generic/icon/icon.scss (hunk 4 above)
Comment (Amber-Song) [reply component-library/components/generic/icon/icon.scss]: Thanks.

File: component-library/components/generic/icon/icon.scss
Comment (Amber-Song) [reply component-library/components/generic/icon/icon.scss:1]: Cool.

File: component-library/components/generic/icon/icon.eleventy.liquid
Comment (Amber-Song) [reply component-library/components/generic/icon/icon.eleventy.liquid:1]: I tried on --brand-grey-900 and it works properly.

File: component-library/components/generic/icon/icon.bookshop.yml (hunk 3 above)
Comment (Amber-Song) [reply component-library/components/generic/icon/icon.bookshop.yml]: Yes. I will add it in the ticket.

**Comment by Amber-Song** [comment]
> The component works well 👍
> 
> I've added a couple of small suggestions. And there is a merge conflict that needs to be resolved.

Yes. I have already changed those.

**Comment by Amber-Song** [comment]
> just a few small changes to do and then we should be able to get this merged.

Done! Thanks for the review. I learnt a lot.

**Review by aleksandrgordeev — APPROVED** [review]
Nice work 👍

File: component-library/components/generic/icon/icon.scss
Comment (James-cloudcannon) [reply component-library/components/generic/icon/icon.scss:1]: Sorry, should of made this one clearer! The above sizes should also include the padding. Currently the svg is set to 32x32px and 69x69px with the addition of 16px padding.

Comment (Amber-Song) [reply component-library/components/generic/icon/icon.scss:1]: Are we using rem or px?

**Comment by AlyshaNolan** [comment]
Made a couple of changes for this one:
- The classes in the top level div were using __ when they should've been using --
- Change big_icon input name to be more editor friendly
- Change implementation for icon color when in bookshop so it can be seen
- Change color input defaults to use hex codes
- Fix styles not being generated on the live site
